packages/main/src/plugin/kube-generator-registry.ts

```diff
@@ -0,0 +1,70 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import { Disposable } from './types/disposable.js';
+
+export type KubernetesGeneratorType = 'Compose' | 'Pod' | 'Container';
+
+export type KubernetesGeneratorSelector = KubernetesGeneratorType | ReadonlyArray<KubernetesGeneratorType>;
+
+export interface GenerateKubeResult {
+  yaml: string;
+}
+
+export interface KubernetesGeneratorProvider {
+  id: string;
+  name: string;
+  generate(engineId: string, ids: string[]): GenerateKubeResult;
+  types: KubernetesGeneratorSelector;
+}
+
+export type KubeGeneratorsInfo = Omit<KubernetesGeneratorProvider, 'generate'>;
```

Comment:
I think it's risky to omit fields

Also the info object needs to be in a separate file as it's imported from the client side

It ensures client will not fetch unwated dependencies if it imports this file